### ISSUE TITLE
Crash on plan: unexpected EOF (Azure RM)

### ISSUE THREAD

**Issue by ben-nz** [hashicorp/terraform#7108]
Hi folks,

Getting a crash when attempting to plan for a deployment to Azure RM. I've successfully deployed to Azure RM previously, this crash is occuring after a re-factor (added variables, split out configs for different resource groups).

### Terraform Version
0.6.16

### Terraform Configuration Files
```hcl
####### File 1

variable "region" {
    type = "string"
    default = "East US"
}

variable "owner" {
    type = "string"
    default = "benparry"
}

variable "environment" {
    type = "string"
    default = "testing"
}

variable "storage_account_type" {
    type = "string"
    default = "Standard_LRS"
}

provider "azurerm" {
  #client_secret is provided at runtime via environment variable: SET ARM_CLIENT_SECRET="<secret>"
  subscription_id = "<redacted>"
  client_id       = "<redacted>"
  tenant_id       = "<redacted>"
}

####### File 2
resource "azurerm_resource_group" "fxcozus1tgenrg020" {
    name = "fxcozus1tgenrg020"
    location = "${var.region}"
    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-adjoin"
    }
}

resource "azurerm_virtual_network" "fxcozus1tgenvn020" {
    name = "fxcozus1tgenvn020"
    address_space = ["10.20.0.0/16"]
    location = "${var.region}"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg020.name}"
    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-adjoin"
    }
}

resource "azurerm_subnet" "fxcozus1tgensn020" {
    name = "fxcozus1tgensn020"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg020.name}"
    virtual_network_name = "${azurerm_virtual_network.fxcozus1tgenvn020.name}"
    address_prefix = "10.20.0.0/24"
    #route_table_id = "${azurerm_route_table.fxcozus1tgenrt020.id}"
}

resource "azurerm_storage_account" "fxcozus1tgensa020" {
    name = "fxcozus1tgensa020"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg020.name}"
    location = "${var.region}"
    account_type = "${var.storage_account_type}"

    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-adjoin"
    }
}

resource "azurerm_storage_container" "vhds" {
    name = "vhds"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg020.name}"
    storage_account_name = "${azurerm_storage_account.fxcozus1tgensa020.name}"
    container_access_type = "private"
}

####### File 3
resource "azurerm_resource_group" "fxcozus1tgenrg021" {
    name = "fxcozus1tgenrg021"
    location = "${var.region}"
    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-jenkinsbuild"
    }
}

resource "azurerm_virtual_network" "fxcozus1tgenvn021" {
    name = "fxcozus1tgenvn021"
    address_space = ["10.21.0.0/16"]
    location = "${var.region}"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg021.name}"
    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-jenkinsbuild"
    }
}

resource "azurerm_subnet" "fxcozus1tgensn021" {
    name = "fxcozus1tgensn021"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg021.name}"
    virtual_network_name = "${azurerm_virtual_network.fxcozus1tgenvn021.name}"
    address_prefix = "10.21.0.0/24"
    #route_table_id = "${azurerm_route_table.fxcozus1tgenrt021.id}"
}

resource "azurerm_storage_account" "fxcozus1tgensa021" {
    name = "fxcozus1tgensa021"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg021.name}"
    location = "${var.region}"
    account_type = "${var.storage_account_type}"

    tags {
        environment = "${var.environment}"
        owner = "${var.owner}"
        project = "chef-jenkinsbuild"
    }
}

resource "azurerm_storage_container" "vhds021" {
    name = "vhds021"
    resource_group_name = "${azurerm_resource_group.fxcozus1tgenrg021.name}"
    storage_account_name = "${azurerm_storage_account.fxcozus1tgensa021.name}"
    container_access_type = "private"
}


```

### Crash and debug logs
[https://gist.github.com/ben-nz/09fdf5806ab60d6f65712db21eec5f67](url)

### Steps to Reproduce
`terraform plan'

**Comment by phinze** [comment]
Relevant lines from crash:

```
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: panic: runtime error: invalid memory address or nil pointer dereference
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: [signal 0xc0000005 code=0x0 addr=0x0 pc=0x4ca727]
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: 
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: goroutine 118 [running]:
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: panic(0xd6c0a0, 0xc082002090)
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: 	/opt/go/src/runtime/panic.go:481 +0x3f4
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: github.com/hashicorp/terraform/builtin/providers/azurerm.resourceArmVirtualNetworkRead(0xc0823083c0, 0xda75c0, 0xc08229e000, 0x0, 0x0)
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/azurerm/resource_arm_virtual_network.go:164 +0x9b7
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc08226ba40, 0xc082201710, 0xda75c0, 0xc08229e000, 0xc08211ada0, 0x0, 0x0)
2016/06/10 09:35:38 [DEBUG] terraform-provider-azurerm.exe: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:210 +0x441
```

**Comment by stack72** [comment]
@ben-nz

Sorry you have been affected by this bug. I have been trying to recreate this now and can't 

* Terraform plan works as expected
* Terraform apply works as expected
* Terraform plan then shows no changes (works as expected)

There were major changes introduced in 0.7 for AzureRM support. I believe this may be fixed. Are you in a situation to try and build of master for this?

Thanks

Paul

**Comment by tasquith** [comment]
+1 I occasionally get this on 0.6.16 too. I'll try taking a build from master

**Comment by stack72** [comment]
Hi @tasquith 

Any thoughts as to whether this has been fixed in 0.7? I still cannot replicate this

Thanks

Paul

**Comment by tasquith** [comment]
@stack72 

I've not had a chance to try with 0.7 yet, however my colleague is now doing some work on Azure - @paradoxbound  

Jim - would it be possible for you to look into this as you've just had the error on 0.6.16

Cheers,

Tom

**Comment by paradoxbound** [comment]
+1 This is hitting me too.

**Comment by stack72** [comment]
Hi All

I have just tried again to replicate this on master and cannot - I believe it is fixed in 0.7 upcoming release. If it isn't, then please do let me know 

Paul
